### PR TITLE
fix: parameter conversion for v3

### LINKED ISSUE #110: Support converting to 3.0.0
It's a long term feature request. 

When we will have the `3.0.0` specification stabilized we should start writing functionality for converting versions `2.X.X` to `3.0.0`.

One of the problems I see are the external sources (local files and http links) which we are not converting at the moment but we should (see issue https://github.com/asyncapi/converter-js/issues/90). In order not to cause problems for people and increase DX we should figure out a way to convert also external files in which messages (and other reusable components) can be defined.

### Scope
- [x] #188
- [ ] https://github.com/asyncapi/converter-js/issues/90
- [x] https://github.com/asyncapi/converter-js/issues/106
- [x] https://github.com/asyncapi/converter-js/issues/107

### PR DESCRIPTION
**Pull request by jonaslagoni**
**Description**
This PR fixes the last problem, parameter conversion to the new object that was left over from #180 

**Related issue(s)**
Fixes #110

### CODE REVIEW

File: src/third-version.ts [hunk 1]
```diff
@@ -356,8 +361,57 @@ function convertComponents(asyncapi: AsyncAPIDocument, options: RequiredConvertV
...
+  const v2ParameterObjectProperties = ["location", "schema", "description", "$ref"];
+  const v2ParameterObjectExtensions = Object.entries(parameter).filter(([key,]) => {
+    return !v2ParameterObjectProperties.includes(key);
+  });
 
+  if(ref !== null) {
+    return {
+      $ref: ref
+    }
+  }
```
Comment (magicmatatjahu) [inline src/third-version.ts:404]:
```suggestion
function convertParameter(parameter: any): any {
  const ref = parameter['$ref'] ?? null;
  if(ref !== null) {
    return {
      $ref: ref
    }
  }

  const enumValues = parameter.schema?.enum ?? null;
  const defaultValues = parameter.schema?.default ?? null;
  const description = parameter.description ?? parameter.schema?.description ?? null;
  const examples = parameter.schema?.examples ?? null;
  const location = parameter.location ?? null;

  //Make sure we keep parameter extensions
  const v2ParameterObjectProperties = ["location", "schema", "description"];
  const v2ParameterObjectExtensions = Object.entries(parameter).filter(([key,]) => {
    return !v2ParameterObjectProperties.includes(key);
  });
```

**Review by derberg — APPROVED** [review]

File: src/third-version.ts [hunk 2]
```diff
@@ -356,8 +361,57 @@ function convertComponents(asyncapi: AsyncAPIDocument, options: RequiredConvertV
...
+
+  //Return the new v3 parameter object
+  return Object.assign({...v2ParameterObjectExtensions},
+    enumValues === null ? null : {enum: enumValues},
+    defaultValues === null ? null : {default: defaultValues},
+    description === null ? null : {description},
+    examples === null ? null : {examples},
+    examples === null ? null : {location}
```
Comment (magicmatatjahu) [inline src/third-version.ts:412]:
```suggestion
    location === null ? null : {location}
```

**Review by magicmatatjahu — CHANGES_REQUESTED** [review]

**Comment by jonaslagoni** [comment]
Nice catch @magicmatatjahu 😆

**Comment by magicmatatjahu** [comment]
I wonder how to treat all stuff like `minLength` etc in schema, as extension with `x-` prefix or skip/leave it?

**Comment by jonaslagoni** [comment]
> I wonder how to treat all stuff like minLength etc in schema, as extension with x- prefix or skip/leave it?

Lets make the users decide that, see if anyone complains down the road.

If we receive complains, then it might be that it should be included in the spec as well, not explicitly in the converter. Because that means users use them, when we assumed they did not.

**Comment by jonaslagoni** [comment]
Also an option yea 👍 

But the spec assumes these properties are not used at all (why they are not present in the new parameter object), so I don't think it makes sense to assume it here in the tool 🙂

**Comment by jonaslagoni** [comment]
Mind making the final review @magicmatatjahu? 🙏

**Review by magicmatatjahu — APPROVED** [review]
LGTM!

**Comment by jonaslagoni** [comment]
Thank you @magicmatatjahu 🙇

**Pull request merged**